client/java/generator/src/main/java/io/openlineage/client/SchemaParser.java

```diff
@@ -48,9 +49,14 @@ public Type parse(JsonNode typeJson) {
             JsonNode properties = typeJson.get("properties");
             for (Iterator<Entry<String, JsonNode>> fieldsJson = properties.fields(); fieldsJson.hasNext(); ) {
               Entry<String, JsonNode> field = fieldsJson.next();
-              Type fieldType = parse(field.getValue());
               String description = field.getValue().has("description") ? field.getValue().get("description").asText() : null;
-              fields.add(new Field(field.getKey(), fieldType, description));
+              if (field.getValue().has("const")) {
+                // only String const are supported
+                fields.add(new Field(field.getKey(), new PrimitiveType("string", null), description, field.getValue().get("const").asText()));
```

Comment:
to avoid surprises maybe we should still call parse() but check the type is String and throw an exception if not.
With this implementation, one can put whatever in the type and it will be replaced by string.
This could lead to some hair pulling on unexpected behavior.

Reply:
Makes sense, we should definitely avoid hair pulling. [Second commit](https://github.com/OpenLineage/OpenLineage/pull/4008/commits/3eed2aaff83231c65990acbe6b8f8fc400a35395) should resolve that.